src/OrleansRuntime/GrainDirectory/AdaptiveDirectoryCacheMaintainer.cs

```diff
@@ -14,14 +13,19 @@ internal class AdaptiveDirectoryCacheMaintainer<TValue> : AsynchAgent
 
         private readonly AdaptiveGrainDirectoryCache<TValue> cache;
         private readonly LocalGrainDirectory router;
+        private readonly Func<List<ActivationAddress>, TValue> updateFunc;
```

Comment:
Update function is not a great name, but can't think of a better one.
This pattern, over all, is awful, but not sure how to fix it, atm. :/
No change requested..  just making noises as bits of my soul die.